### PR TITLE
Feature/signal class

### PR DESCRIPTION
**Pull request by jajcayn**
for now just the class itself, please have a look and possibly comment on capabilities; i.e. DO NOT MERGE now please, just after I add tests

**to be added:**
* [x] tests for `Signal` class
* [x] small notebook showcasing the `Signal` capabilities

### CODE REVIEW

File: neurolib/utils/signal.py [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+        "name",
+        "label",
+        "signal_type",
+        "unit",
+        "description",
+        "process_steps",
+    ]
+
+    @classmethod
+    def from_file(cls, filename):
```
Comment (caglorithm) [inline neurolib/utils/signal.py]: What's the use case scenario of this? Is this so the user can save and load processed signals to disk? Should we have this for model outputs as well?

File: neurolib/utils/signal.py [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+        """
+        Load signal from saved file.
+
+        :param filename: filename for the Signal
+        :type filename: str
+        """
+        if not filename.endswith(NC_EXT):
+            filename += NC_EXT
+        # load NC file
+        xarray = xr.load_dataarray(filename)
```
Comment (caglorithm) [inline neurolib/utils/signal.py]: Maybe we should adopt a unified way of saving arrays / signals / etc. This sounds interesting, if we adopt `xarray` throughout the project.

File: neurolib/utils/signal.py [hunk 3]
```diff
@@ -0,0 +1,515 @@
...
+Base classes for representing signals.
+"""
+
+import logging
+from copy import deepcopy
+
+import numpy as np
+import xarray as xr
+from mne import set_log_level
+from mne.filter import filter_data, resample
```
Comment (caglorithm) [inline neurolib/utils/signal.py]: If you could do this without mne but with scipy.filter (which we already require) you'd be the mvp!

File: requirements.txt [hunk 4]
```diff
@@ -8,3 +8,4 @@ pandas
 tqdm
 pypet
 deap
+mne
```
Comment (caglorithm) [inline requirements.txt]: 💻🔥

File: neurolib/utils/signal.py (hunk 1 above)
Comment (jajcayn) [reply neurolib/utils/signal.py]: typically you'd do something like

```python
signal = RatesSignal.from_file("filename_of_saved_data.nc")
```

which is shorter and more concise than something like:
```
array = xr.load_dataarray("filename_of_saved_data.nc")
signal = RatesSignal(data=array)
```

**Comment by caglorithm** [comment]
LGTM! Do you wanna merge after making the notebook or before?

**Comment by jajcayn** [comment]
> LGTM! Do you wanna merge after making the notebook or before?

after pls, I am in the process of making it, just be approx 30 more mins)

**Comment by caglorithm** [comment]
> > LGTM! Do you wanna merge after making the notebook or before?
> 
> after pls, I am in the process of making it, just be approx 30 more mins)

Take your time pls, no hurry :)

**Comment by jajcayn** [comment]
@caglorithm all done, after the tests run and you check the notebook, we can merge :)

**Comment by caglorithm** [comment]
Congrats on a very nice and extensive notebook! I just found some minor errors here and there but commenting on a notebook is not easy here on github so let's just accept it and correct it later! 

Merging!

**Pull request merged**